test/contracts/host-contracts.mjs

```
@@ -0,0 +1,86 @@
+async function sha256Hex(input) {
+  const data = typeof input === 'string' ? te(input) : te(JSON.stringify(input));
+  const d = await globalThis.crypto.subtle.digest('SHA-256', data);
+  return [...new Uint8Array(d)].map(b => b.toString(16).padStart(2, '0')).join('');
+}
+
+function byteLen(str) { return te(str).length; }
+
+function sanitizeGraphQL(sdl, maxBytes = 5 * 1024 * 1024) {
+  if (typeof sdl !== 'string') throw new Error('Schema must be a string');
+  if (byteLen(sdl) > maxBytes) { const e = new Error(`Schema exceeds max size (${maxBytes} bytes)`); e.code = 'EINPUTSIZE'; throw e; }
+  return sdl.replace(/^\uFEFF/, '').replace(/\u0000/g, '');
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)